### PR TITLE
[relates #281] DB verifying steps to scenarios. part 1

### PR DESCRIPTION
**Pull request by Stefan365**
DB veryfying steps, part 1

### CODE REVIEW

File: utilities/src/main/java/io/syndesis/qe/utils/DbUtils.java [hunk 1]
```diff
@@ -75,4 +103,59 @@ public void deleteRecordsInTable(String tableName) {
 			log.error("Error: " + ex);
 		}
 	}
+
+	public Integer insertInDB(String tableName, List<List<String>> dataTable) {
```
Comment (dsimansk) [inline utilities/src/main/java/io/syndesis/qe/utils/DbUtils.java]: I wonder if the same result could be achieve by `executeSqlOnSampleDb()` and processing the result outside DbUtil class. This class should stay pretty genetic and the testing logic executed in verifying steps.

cc @tplevko wdyt?

File: utilities/src/main/java/io/syndesis/qe/utils/DbUtils.java [hunk 2]
```diff
@@ -75,4 +103,59 @@ public void deleteRecordsInTable(String tableName) {
...
+			int newId = st.executeUpdate();
+			log.info("***NEW INT: *{}*", newId);
+			return newId;
+		} catch (SQLException e) {
+			log.error("SQL ERROR: " + e);
+		}
+		return null;
+	}
+
+	public List<Integer> getCompletedByTask(String task) {
```
Comment (dsimansk) [inline utilities/src/main/java/io/syndesis/qe/utils/DbUtils.java]: Same as for method above.

Comment (Stefan365) [reply utilities/src/main/java/io/syndesis/qe/utils/DbUtils.java]: @dsimansk I have feeling construction of SQL query should be part of DbUtils, WDYT?

Comment (dsimansk) [reply utilities/src/main/java/io/syndesis/qe/utils/DbUtils.java]: As I've described in the other comment. DbUtils should be more like entry point how to retrieve desired data for a test, not necessary hide whole validation logic etc. With a retrieved ResultSet you have more flexibility to assert all kind of values when needed.

File: ui-tests/src/test/java/io/syndesis/qe/pages/integrations/detail/IntegrationDetailPage.java [hunk 3]
```diff
@@ -19,8 +19,8 @@
 	public final List<String> actionsSet = new ArrayList<String>();
 
 	private static final class Element {
-		public static final By ROOT = By.cssSelector("syndesis-integration-detail-page");
-		public static final By STATUS = By.cssSelector("syndesis-integration-status");
+		public static final By ROOT = By.cssSelector("syndesis-integrations-detail-page");
+		public static final By STATUS = By.cssSelector("syndesis-integrations-status");
```
Comment (lvydra) [inline ui-tests/src/test/java/io/syndesis/qe/pages/integrations/detail/IntegrationDetailPage.java]: Are you sure with these? I'm not sure if we are using same version but i got integration without s.

Comment (Stefan365) [reply ui-tests/src/test/java/io/syndesis/qe/pages/integrations/detail/IntegrationDetailPage.java]: in TP3 it is again with 's', (but not everywhere) see scenario @db-connection-crud-1-read-update comment

Comment (lvydra) [reply ui-tests/src/test/java/io/syndesis/qe/pages/integrations/detail/IntegrationDetailPage.java]: OK, probably it will be best to wait.

Comment (dsimansk) [reply ui-tests/src/test/java/io/syndesis/qe/pages/integrations/detail/IntegrationDetailPage.java]: I can see singular variant without `s`. I'd rather exclude it from PR for now and pls double-check the deployed version.

File: utilities/src/main/java/io/syndesis/qe/bdd/validation/DbValidationSteps.java [hunk 4]
```diff
@@ -68,13 +74,53 @@ public void validateLead() {
 		Assertions.assertThat(getLeadTaskFromDb().toLowerCase()).isNotEmpty();
 	}
 
+	@Then("^validate add_lead procedure with last_name: \"([^\"]*)\", company: \"([^\"]*)\", period in ms: \"(\\w+)\"")
+	public void validateAddLeadProcedure(String lastName, String company, Integer ms) {
+		//wait for period cycle:
+		Selenide.sleep(ms + 100);
```
Comment (dsimansk) [inline utilities/src/main/java/io/syndesis/qe/bdd/validation/DbValidationSteps.java]: We definitely don't need Selenide's sleep, as it brings whole dependency. `Thread.sleep` or `TimeUnit.SECONDS.sleep` is good enough. If you mind the `InterruptedException` add something like sleepQuietly() into `TestUtils` class.

File: utilities/src/main/java/io/syndesis/qe/bdd/validation/DbValidationSteps.java [hunk 5]
```diff
@@ -68,15 +71,96 @@ public void validateLead() {
...
+	@Then("^validate that all todos with task \"([^\"]*)\" have value completed \"(\\w+)\", period in ms: \"(\\w+)\"$")
+	public void checksThatAllTodosHaveCompletedVal(String task, Integer val, Integer ms) throws InterruptedException {
+		Thread.sleep(ms + 1000);
+
+		ResultSet rs;
+		List<Integer> completedAll = new ArrayList<>();
+		try {
+			String sql = String.format("SELECT completed FROM todo WHERE task like %s", task);
+			rs = dbUtils.readSqlOnSampleDb(sql);
+			while (rs.next()) {
```
Comment (dsimansk) [inline utilities/src/main/java/io/syndesis/qe/bdd/validation/DbValidationSteps.java]: I think this whole check can be simplified to 

```
Assertions.assertThat(rs).isEqualToComparingOnlyGivenFields(val, "completed");
```

File: utilities/src/main/java/io/syndesis/qe/bdd/validation/DbValidationSteps.java [hunk 6]
```diff
@@ -68,15 +71,96 @@ public void validateLead() {
...
+				Integer compl = rs.getInt("completed");
+				completedAll.add(compl);
+			}
+			for (Integer compl : completedAll) {
+				Assertions.assertThat(compl).isEqualTo(val);
+			}
+		} catch (SQLException ex) {
+			log.error("Error: " + ex);
```
Comment (dsimansk) [inline utilities/src/main/java/io/syndesis/qe/bdd/validation/DbValidationSteps.java]: I'd also add Assertions.fail() upon SQL error.

File: ui-tests/src/test/java/io/syndesis/qe/pages/integrations/detail/IntegrationDetailPage.java (hunk 3 above)
Comment (Stefan365) [reply ui-tests/src/test/java/io/syndesis/qe/pages/integrations/detail/IntegrationDetailPage.java]: ok

File: utilities/src/main/java/io/syndesis/qe/bdd/validation/DbValidationSteps.java (hunk 5 above)
Comment (Stefan365) [reply utilities/src/main/java/io/syndesis/qe/bdd/validation/DbValidationSteps.java]: ok

**Pull request merged**